examples/compton/plot_Ebalance.py

```diff
@@ -0,0 +1,187 @@
+Nt_init = 2         # number of time steps
+
+Tmax_restart_ioniz = 2e-6
+Nt_restart_ioniz = 500
+
+Tmax_restart_eq = 30e-3
+Nt_restart_eq = 1000
+
+Tmax_restart_rad=1e-11
+Nt_restart_rad=2
+
+Nr = 151             # number of radial grid points
+times  = [0]        # times at which parameters are given
+radius = [0, 2]     # span of the radial grid
+radialgrid = np.linspace(radius[0],radius[-1],Nr)
+radius_wall = 2.15  # location of the wall 
+
+E_initial = 0.001 # initial electric field in V/m
+E_wall = 0.0001        # boundary electric field in V/m
+# NOTE: it does not work to have self-consistent E-field with prescribed BC with E_wall=0, 
+# since that leads to Psi_wall=0 constantly, which does not work when you have a relative tolerance
+
+T_initial = np.logspace(np.log10(0.7),np.log10(2e3),Nr)    # initial temperature in eV
+
+# Set up radial grid
+ds.radialgrid.setB0(B0)
+ds.radialgrid.setMinorRadius(radius[-1])
+ds.radialgrid.setNr(Nr)
+
+# Set time stepper
+ds.timestep.setTmax(Tmax_init)
+ds.timestep.setNt(Nt_init)
+
+# Set ions
+Z0=1
+Z=10
+
+"""
+n_D_tmp=np.zeros(2)
+n_D_tmp[0]=0*n_D
+n_D_tmp[1]=1*n_D
+n_D_tmp=n_D_tmp.reshape(-1,1)*np.ones((1,len(radius)))
+ds.eqsys.n_i.addIon(name='D', Z=1, iontype=Ions.IONS_DYNAMIC, n=n_D_tmp,r=np.array(radius))
+
+n_Z_tmp=np.zeros(Z+1)
+n_Z_tmp[Z0]=n_Z
+n_Z_tmp=n_Z_tmp.reshape(-1,1)*np.ones((1,len(radius)))
+ds.eqsys.n_i.addIon(name='Ar', Z=Z, iontype=Ions.IONS_DYNAMIC, n=n_Z_tmp,r=np.array(radius))
+"""
+ds.eqsys.n_i.addIon(name='D', Z=1, iontype=Ions.IONS_DYNAMIC_FULLY_IONIZED, n=n_D)
+ds.eqsys.n_i.addIon(name='Ne', Z=Z, iontype=Ions.IONS_DYNAMIC_NEUTRAL, n=n_Z)
+
+# ds.eqsys.n_i.addIon(name='D', Z=1, iontype=Ions.IONS_EQUILIBRIUM, n=n_D)
+# ds.eqsys.n_i.addIon(name='Ar', Z=Z, iontype=Ions.IONS_EQUILIBRIUM, n=n_Z)
+
+temperature = T_initial * np.ones((len(times), len(radialgrid)))
+ds.eqsys.T_cold.setPrescribedData(temperature=temperature, times=times, radius=radialgrid)
+
+# Set E_field 
+efield = E_initial*np.ones((len(times), len(radius)))
+ds.eqsys.E_field.setPrescribedData(efield=efield, times=times, radius=radius)
+ds.eqsys.E_field.setBoundaryCondition(wall_radius=radius_wall)
+
+# Set runaway generation rates
+# ds.eqsys.n_re.setCompton(RE.COMPTON_RATE_ITER_DMS)
+# ds.eqsys.n_re.setAvalanche(RE.AVALANCHE_MODE_FLUID)
+
+# Disable runaway and hot-tail grid
+ds.runawaygrid.setEnabled(False)
+ds.hottailgrid.setEnabled(False)
+
+# ds.solver.setType(Solver.LINEAR_IMPLICIT)
+# Use the nonlinear solver
+ds.solver.setType(Solver.NONLINEAR)
+
+ds.solver.setLinearSolver(linsolv=Solver.LINEAR_SOLVER_LU)
+ds.solver.setTolerance(reltol=0.01)
+ds.solver.setMaxIterations(maxiter = 500)
+#ds.solver.setVerbose(True)
+
+
+ds.other.include('fluid', 'lnLambda','nu_s','nu_D')
+
+
+# Save settings to HDF5 file
+ds.save('init_settings.h5')
+runiface(ds, 'output_init.h5', quiet=False)
+#### Ionization #############
+ds2 = DREAMSettings(ds)
+
+ds2.fromOutput('output_init.h5')
```

Comment:
If you have called the ds.output.setFilename(…) function before saving the settings object, this line is not needed. As it is now, it is however possible to run the simulation using runiface without doing that (the output filename is instead specified in the call to runiface), and then DREAMSettings(ds) calls fromOutput with the default filename 'output.h5'. Wouldn't it be better if runiface used the filename set to the settings object?

Reply:
Ok, that (kind of) makes sense. The ``runiface`` function was written before the output filename could be specified in the ``DREAMSettings`` object. I agree that there should be more interaction between ``runiface`` and the given settings object, although I would argue that it could still be logical for ``runiface`` *not* to load the filename from the settings object. This is because I think it makes sense to be able to run DREAMi easily without saving the output data to disk. However, it would still make sense for ``runiface`` to update the ``ds.output.filename`` option when the output filename is specified as a parameter to ``runiface``. I think this would allow us to remove line 149.